cpp/src/arrow/array-binary-test.cc

```diff
@@ -42,6 +42,30 @@ using internal::checked_cast;
 // ----------------------------------------------------------------------
 // String / Binary tests
 
+template <typename ArrayType>
+void CheckStringArray(const ArrayType& array, const std::vector<std::string>& strings,
+                      const std::vector<uint8_t>& is_valid, int repeats = 1) {
```

Comment:
repeats seems unused. Maybe `ASSERT_EQ(length, base_length * repeats)`, though, not really testing the implementation but the test implementation.

Reply:
Ah, I should add that indeed.